### ISSUE TITLE
Heredocs and escaped \n

### ISSUE THREAD

**Issue by marcandre** [jruby/jruby#6258]
```sh
$ ruby -v
jruby 9.3.0.0-SNAPSHOT (2.6.5) 2020-06-02 ed8cf08770 Java HotSpot(TM) 64-Bit Server VM 25.102-b14 on 1.8.0_102-b14 +jit [darwin-x86_64]
$ cat temp.rb
puts <<~TEST
  hello, \
  world!
TEST
$ ruby temp.rb
hello,   world!
```

Output has two extra spaces.

**Comment by marcandre** [comment]
Note: Ruby 2.3 has the same output, actually. Ruby 2.4+ has the (IMO correct) output

**Comment by enebo** [comment]
This is fixed on the no_sourceposition which will be merged soon for 9.3.
